lib/samples/set_basemap/set_basemap_sample.dart

```diff
@@ -115,33 +120,23 @@ class _SetBasemapSampleState extends State<SetBasemapSample>
     );
   }
 
+  void onMapViewReady() {
+    // Set the map view controller's map to the ArcGIS map.
+    _mapViewController.arcGISMap = _arcGISMap; 
+  }
+
   void updateMap(Basemap basemap) {
-    _arcGISMap.basemap = basemap;
-    _mapViewController.setViewpointAnimated(_sanFranciscoViewpoint);
-    _mapViewController.setViewpointRotation(angleDegrees: 0.0);
+    // Update the map view with the selected basemap.
+    _arcGISMap.basemap = basemap; 
   }
 
   Future loadBasemaps() async {
-    // load basemaps from online items
-    List<Basemap> basemaps = List.from([
-      Basemap.withUri(Uri.parse(
-          'https://runtime.maps.arcgis.com/home/item.html?id=9b39104916614f0899993934d2f1d375')), // newspaper - from different org
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=358ec1e175ea41c3bf5c68f0da11ae2b')), // dark gray canvas
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=979c6cc89af9449cbeb5342a439c6a76')), // light gray canvas
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=fae788aa91e54244b161b59725dcbb2a')), // OSM
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=28f49811a6974659988fd279de5ce39f')), // Imagery
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=2e8a3ccdfd6d42a995b79812b3b0ebc6')), // Outdoor
-      Basemap.withUri(Uri.parse(
-          'https://www.arcgis.com/home/item.html?id=7e2b9be8a9c94e45b7f87857d8d168d6')), // Streets night
-      Basemap.withStyle(BasemapStyle.arcGISNavigation), // No thumbnail
-    ]);
+    // Create a portal to access online items.
+    final portal = Portal.arcGISOnline();
+    // Load basemaps from portal.
+    final basemaps = await portal.developerBasemaps();
 
-    // load each basemap to access and display attribute data in the UI
+    // Load each basemap to access and display attribute data in the UI.
     for (var basemap in basemaps) {
```

Comment:
This could be `for (final basemap in basemaps) {`
And below `var thumbnail =` can be `final thumbnail =`